android/src/main/java/com/reactnativecommunity/netinfo/NetInfoModule.java

```diff
@@ -15,10 +15,11 @@
 
 /** Module that monitors and provides information about the connectivity state of the device. */
 @ReactModule(name = NetInfoModule.NAME)
-public class NetInfoModule extends ReactContextBaseJavaModule {
+public class NetInfoModule extends ReactContextBaseJavaModule implements AmazonFireDeviceConnectivityPoller.ConnectivityChangedCallback {
     public static final String NAME = "RNCNetInfo";
 
     private final ConnectivityReceiver mConnectivityReceiver;
+    private final AmazonFireDeviceConnectivityPoller mAmazonConnectivityChecker;
```

Comment:
I think since work is customizing how connectivity events are handled for FireOS devices, your `AmazonFireDeviceConnectivityPoller` should implement the `ConnectivityReceiver` interface and be used instead of `BroadcastReceiverConnectivityReceiver` on Android devices.
The pattern of having device-specific receivers as separate objects will start to have scalability issues if more device-specific customizations are required.

Reply:
That was my first instinct as well, but we still want the code or what the receivers do, with a customization in the way reachability is checked. That means that there would need to be two implementations of the amazon one (one for API levels below 24 and one above) using factored out logic for the amazon specific poller. I could do it that way if people insist, but I am also open to other approaches on how to inject customizations of the reachability check of the existing receivers.